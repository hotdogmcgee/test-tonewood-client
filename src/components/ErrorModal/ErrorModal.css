.modal {
  position: fixed;
  top: 0;
  left: 0;
  display: block;
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.6);
}

.modal-main {
  position: fixed;
  background: white;
  width: 80%;
  height: auto;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

.display-block {
  display: block;
}

.display-none {
  display: none;
}

.error-message-container {
  margin-bottom: 10px;
  padding: 20px;
  text-align: center;
}
