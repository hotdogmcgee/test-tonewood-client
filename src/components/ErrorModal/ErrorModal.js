import React from "react";
import "./ErrorModal.css";
import { Button } from "../../components/Utils/Utils";

export function ErrorModal(props) {
  const showHideClassName = props.show
    ? "modal display-block"
    : "modal display-none";
  return (
    <div className={showHideClassName}>
      <section className="modal-main error-modal">
        <div className="error-message-container">{props.children}</div>

        <Button onClick={props.handleClose}>Close</Button>
      </section>
    </div>
  );
}
