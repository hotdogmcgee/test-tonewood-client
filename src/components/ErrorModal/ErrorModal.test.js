import React from "react";
import { ErrorModal } from "./ErrorModal";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("ErrorModal component", () => {
  const children = "there was an error";

  it("renders a ErrorModal by default", () => {
    const wrapper = shallow(<ErrorModal />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the ErrorModal given props", () => {
    const wrapper = shallow(<ErrorModal>{children}</ErrorModal>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
