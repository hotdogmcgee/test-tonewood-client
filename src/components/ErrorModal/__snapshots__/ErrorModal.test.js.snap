// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorModal component renders a ErrorModal by default 1`] = `
<div
  className="modal display-none"
>
  <section
    className="modal-main error-modal"
  >
    <div
      className="error-message-container"
    />
    <Button>
      Close
    </Button>
  </section>
</div>
`;

exports[`ErrorModal component renders the ErrorModal given props 1`] = `
<div
  className="modal display-none"
>
  <section
    className="modal-main error-modal"
  >
    <div
      className="error-message-container"
    >
      there was an error
    </div>
    <Button>
      Close
    </Button>
  </section>
</div>
`;
